packages/vue-flicking/src/Flicking.ts

```diff
@@ -102,43 +145,114 @@ export default class Flicking extends Vue {
+        // const panel = this.$_getPanelFromRelativeIndex(index, allPanels);
+        let node: VNode;
+        if (index < 0) {
+          const relativeIndex = panelCnt + ((index + 1) % panelCnt - 1);
+          const cloneIndex = newCloneCount - ((index + 1) % panelCnt + 1);
+          const origSlot = slots[relativeIndex];
+
+          node = this.$_cloneVNode(origSlot, h, cloneIndex, relativeIndex);
+        } else if (index >= panelCnt) {
+          const relativeIndex = index % panelCnt;
+          const origSlot = slots[relativeIndex];
+
+          node = this.$_cloneVNode(origSlot, h, Math.floor(index / panelCnt) - 1, relativeIndex);
+        } else {
+          node = slots[index];
+        }
+
+        return node;
+      });
+
+      const added = slotsDiff.added.map(addedIndex => slots[addedIndex]);
+
+      const addedClones = [...Array(oldCloneCount).keys()].reduce((clones: VNode[], cloneIndex) => {
+        const newAddedClones = slotsDiff.added.map(addedIdx => {
+          const child = slots[addedIdx];
+
+          return this.$_cloneVNode(child, h, cloneIndex, addedIdx);
+        });
+        return [...clones, ...newAddedClones];
+      }, []);
+
+      const newClones = [...Array(newCloneCount - oldCloneCount).keys()].reduce((clones: VNode[], idxOffset) => {
+        const cloneIndex = oldCloneCount + idxOffset;
+        const childs = slots.map((slot, slotIndex) => {
+          return this.$_cloneVNode(slot, h, cloneIndex, slotIndex);
+        });
+
+        return [...clones, ...childs];
+      }, []);
+
+      panels = [...visibles, ...added, ...addedClones, ...newClones];
+    } else {
+      panels = [...slots.slice(0, lastIndex + 1), ...this.$_getClonedVNodes()];
+    }
 
     return panels;
   }
 
   private $_getClonedVNodes() {
     const h = this.$createElement;
-    const cloneCount = this.$_cloneCount;
+    const cloneCount = this.$_nativeFlicking
+      ? this.$_nativeFlicking.getCloneCount()
+      : 0;
     const lastIndex = this.$_nativeFlicking
       ? this.$_nativeFlicking.getLastIndex()
       : this.options.lastIndex || DEFAULT_OPTIONS.lastIndex;
     const children = this.$slots.default!.slice(0, lastIndex + 1);
     const clones: VNode[] = [];
 
     for (let cloneIndex = 0; cloneIndex < cloneCount; cloneIndex++) {
-      const childKeys = children.map((child, childIdx) => child.key ? child.key : childIdx);
-      clones.push(...children.map((child, childIdx) => this.$_cloneVNode(child, h, `clone${cloneIndex}-${childKeys[childIdx]}`)));
+      clones.push(...children.map((child, childIdx) => this.$_cloneVNode(child, h, cloneIndex, childIdx)));
     }
     return clones;
   }
 
-  private $_cloneVNode(vnode: VNode, h: CreateElement, key?: string | number): VNode {
+  private $_cloneVNode(vnode: VNode, h: CreateElement, cloneIndex?: number, childIndex?: number): VNode {
+    const key = cloneIndex != null
+      ? `clone${cloneIndex}-${vnode.key ? vnode.key : childIndex}`
+      : undefined;
```

Comment:
@WoodNeck Is it work although `key` is undefined?

Reply:
It is not essential for the Vue.